src/menus/ExportMenu/ExportMenu.js

```diff
@@ -150,14 +150,16 @@ const ExportMenu = () => {
   const ref = useRef();
 
   useEffect(() => {
-    if (maxCanvasSize) {
-      return () => {};
-    }
-    // Calculate maxCanvasSize on topic load when not already in local storage
-    const size = determineMaxCanvasSize();
-    if (size) {
-      setMaxCanvasSize(size);
-      localStorage.setItem(LS_SIZE_KEY, size);
+    if (!maxCanvasSize) {
+      canvasSize
+        .maxArea({ usePromise: true })
+        .then((result) => {
+          const size = Math.max(result.width, result.height);
+          setMaxCanvasSize(size);
+          localStorage.setItem(LS_SIZE_KEY, size);
+        })
+        // eslint-disable-next-line no-console
+        .catch((result) => console.log('Error', result));
     }
     return () => {};
```

Comment:
Can you abort the calculation on unmount ?

Reply:
Using [cancelable-promise](https://www.npmjs.com/package/cancelable-promise) to cancel on unmount